### DISCUSSION TITLE
Channel update rate for high doppler simulations using TimeChannel

### DISCUSSION THREAD

**Discussion by naperman** [NVlabs/sionna#396] [General]
I'm trying to figure out how to increase the channel update rate for a high doppler scenario(say $\Delta f_{\text{doppler}}$ = 1000Hz).  

```py
import sionna
from sionna.ofdm import ResourceGrid

rg = ResourceGrid(num_ofdm_symbols = 14,
                  fft_size = 64,
                  subcarrier_spacing = 30e3,
                  num_tx = 1,
                  num_streams_per_tx = 1,
                  num_guard_carriers = [5, 6],
                  dc_null = True,
                  pilot_pattern = "kronecker",
                  pilot_ofdm_symbol_indices = [2, 11])
rg.show();
print(f"Bandwidth is: {rg.bandwidth/1e6}MHz")
print(f"Number of time steps: {rg.num_time_samples}")
```

> Bandwidth is: 1.92MHz
> Number of time steps: 896

Now we calculate the time lags from channel filter for this bandwidth

```py
l_min, l_max = time_lag_discrete_time_channel(rg.bandwidth)
l_tot = l_max - l_min + 1
print(f"l_max = {l_max}. l_min = {l_min}, l_tot = {l_tot}")
num_time_steps = rg.num_time_samples + l_tot - 1
print(f"Total number of time steps for TimeChannel Simulation: {num_time_steps}")
```

> l_max = 12. l_min = -6, l_tot = 19
> Total number of time steps for TimeChannel Simulation: 914

Now add a doppler shift componant of $\Delta f_{\text{doppler}}$ = 1000Hz to this bandwidth and check the time lags and total number of time steps 

```py
delta_f_doppler = 1000
l_min, l_max = time_lag_discrete_time_channel(rg.bandwidth + 2*delta_f_doppler)
l_tot = l_max - l_min + 1
print(f"l_max = {l_max}. l_min = {l_min}, l_tot = {l_tot}")
num_time_steps = rg.num_time_samples + l_tot - 1
print(f"Total number of time steps for TimeChannel Simulation: {num_time_steps}")
```

> l_max = 12. l_min = -6, l_tot = 19
> Total number of time steps for TimeChannel Simulation: 914

We see that the number of time steps has changed. 

Numerically, this makes sense as the doppler shift component added to BW is much smaller compared to the BW. 
But, my question is the additional number of time steps introduces by _l_tot_ sufficient to deal with most realistic high doppler scenarios or do we need to do anything more to increase channel update rate to simulate such scenarios?

**Comment by jhoydis** [comment]
Hi @naperman,

The number of time steps is unrelated to the bandwidth/sampling rate. As long as the Doppler shift is small compared to the bandwidth, you can probably ignore the spectral growth in your simulations. However, if this is relevant for you, you would need to simulate the whole system at a larger sampling rate. There is a related discussion item #427.
